packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -2,15 +2,32 @@ import { NextResponse, NextRequest, userAgent } from 'next/server';
 import {
   GraphQLPersonalizeService,
   GraphQLPersonalizeServiceConfig,
-  CdpService,
-  CdpServiceConfig,
-  ExperienceParams,
   getPersonalizedRewrite,
   PosResolver,
 } from '@sitecore-jss/sitecore-jss/personalize';
-import { debug, NativeDataFetcher } from '@sitecore-jss/sitecore-jss';
```

Comment:
@ambrauer I would prefer to leave this implementation in our SDK. In the scope of a "minor" release, we can deprecate only (can be removed in the scope of a major release). But I think it can be useful for some customers who implemented their own middleware and decided to use our _NativeDataFetcher_, can it be a case?

Reply:
Correct, we'd only be able to deprecate until JSS 22. But I agree, let's leave it for now as I can see it being useful for customers and our future selves :)

Reply:
Maybe we will reuse it if we will implement a new middleware as an example, who knows? I assume this functionality will never be broken, since it uses a native function